packages/react/src/publication/__tests__/useNotInterestedToggle.spec.ts

```diff
@@ -1,136 +1,69 @@
-      return renderHookWithMocks(callback, {
-        mocks: {
-          apolloClient: apolloClient,
-          publicationCacheManager: cacheManager,
-        },
-      });
-    },
-
-    get updatedPublicationFragment() {
-      return apolloClient.cache.readFragment({
-        id: apolloClient.cache.identify(publication),
-        fragment,
-        fragmentName: publication.__typename,
-      });
-    },
-  };
-}
+import { usePublication } from '../usePublication';
 
 describe(`Given the ${useNotInterestedToggle.name} hook`, () => {
-  describe('when invoking the hook callback', () => {
-    it.each([
-      {
-        initial: true,
-        expected: false,
-      },
-      {
-        initial: false,
-        expected: true,
-      },
-    ])(
-      'should toggle the `publication.notInterested` from "$initial" to "$expected"',
-      async ({ initial, expected }) => {
-        const publication = mockPostFragment({
-          operations: mockPublicationOperationsFragment({ isNotInterested: initial }),
-        });
-        const scenario = setupTestScenario({
-          fragment: FragmentPost,
-          publication,
-        });
+  const publication = mockPostFragment();
+
+  describe('when calling the execute method', () => {
+    it('should call correct mutation', async () => {
+      const { renderHook } = await setupHookTestScenarioWithSession([
+        mockPublicationResponse({
+          variables: {
+            request: { forId: publication.id },
+          },
+          result: publication,
+        }),
+        mockAddPublicationNotInterestedResponse({
+          request: { on: publication.id },
+        }),
+        mockUndoPublicationNotInterestedResponse({
+          request: { on: publication.id },
+        }),
+      ]);
 
-        const { result } = scenario.renderHook(() => useNotInterestedToggle());
+      const { result: publicationResult } = renderHook(() =>
+        usePublication({ forId: publication.id }),
+      );
+      const { result } = renderHook(() => useNotInterestedToggle());
 
-        act(() => {
-          void result.current.execute({ publication });
-        });
+      // put publication in cache
+      await waitFor(() => expect(publicationResult.current.loading).toBeFalsy());
 
-        expect(result.current).toMatchObject({
-          loading: true,
-        });
+      invariant(publicationResult.current.data, 'publication not found');
+      invariant(isPrimaryPublication(publicationResult.current.data), 'not a primary publication');
 
-        await waitFor(() => expect(result.current.loading).toBe(false));
-        expect(scenario.updatedPublicationFragment).toMatchObject({
-          operations: {
-            isNotInterested: expected,
-          },
-        });
-      },
-    );
+      expect(publicationResult.current.data.operations.isNotInterested).toBe(false);
 
-    it('should update the `publication.mirrorOf.notInterested` in case the `publication` is a Mirror', async () => {
```

Comment:
Is this now a very long test?

Can it be broken down?

Reply:
I did break it, but I didn't notice any time saving in test execution https://github.com/lens-protocol/lens-sdk/pull/687/commits/c554b32718b38e42f02db6fc425dd5a92903a9f7

Reply:
My point was mostly in readability/maintainability. The need to add comments seems a symptom.